src/FlexVotingClient.sol

```diff
@@ -194,19 +194,21 @@ abstract contract FlexVotingClient {
 
   /// @dev Checkpoints the _user's current raw balance.
   function _checkpointRawBalanceOf(address _user) internal {
-    balanceCheckpoints[_user].push(_rawBalanceOf(_user));
+    balanceCheckpoints[_user].push(SafeCast.toUint32(block.number), _rawBalanceOf(_user));
   }
 
   /// @notice Returns the `_user`'s raw balance at `_blockNumber`.
   /// @param _user The account that's historical raw balance will be looked up.
   /// @param _blockNumber The block at which to lookup the _user's raw balance.
   function getPastRawBalance(address _user, uint256 _blockNumber) public view returns (uint256) {
-    return balanceCheckpoints[_user].getAtProbablyRecentBlock(_blockNumber);
+    uint32 key = SafeCast.toUint32(_blockNumber);
+    return balanceCheckpoints[_user].upperLookup(key);
   }
 
   /// @notice Returns the sum total of raw balances of all users at `_blockNumber`.
   /// @param _blockNumber The block at which to lookup the total balance.
   function getPastTotalBalance(uint256 _blockNumber) public view returns (uint256) {
-    return totalBalanceCheckpoints.getAtProbablyRecentBlock(_blockNumber);
+    uint32 key = SafeCast.toUint32(_blockNumber);
```

Comment:
Could we just make the argument a `uint32`?

Reply:
Nevermind.

We could change the argument type, but given that the caller is just passing along `GOVERNOR.proposalSnapshot`, which is a uint256, we'd have to end up casting it in the callers instead. So there'd be no benefit